basyx.aasxfileserver/basyx.aasxfileserver-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/aasxfileserver/AASXFileServerMongoDBBackendProvider.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+* Copyright (C) 2024 the Eclipse BaSyx Authors
+* 
+* Permission is hereby granted, free of charge, to any person obtaining
+* a copy of this software and associated documentation files (the
+* "Software"), to deal in the Software without restriction, including
+* without limitation the rights to use, copy, modify, merge, publish,
+* distribute, sublicense, and/or sell copies of the Software, and to
+* permit persons to whom the Software is furnished to do so, subject to
+* the following conditions:
+* 
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+* NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+* LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+* OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+* WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+* 
+* SPDX-License-Identifier: MIT
+******************************************************************************/
+
+package org.eclipse.digitaltwin.basyx.aasxfileserver;
+
+import org.eclipse.digitaltwin.basyx.aasxfileserver.backend.AASXFileServerBackendProvider;
+import org.eclipse.digitaltwin.basyx.aasxfileserver.model.Package;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of {@link AASXFileServerBackendProvider} for MongoDB Backend
+ *
+ * @author zielstor, fried
+ *
+ */
+@ConditionalOnExpression("'${basyx.backend}'.equals('MongoDB')")
+@Component
+public class AASXFileServerMongoDBBackendProvider implements AASXFileServerBackendProvider {
+
+	private AASXFileServerMongoDBCrudRepository repository;
+
+	@Autowired
+	public AASXFileServerMongoDBBackendProvider(MongoTemplate template, @Value("${basyx.aasx-fileserver.mongodb.collectionName:aasx-fileserver}") String collectionName, @Value("${basyx.aasx-fileserver.mongodb.collectionName:aasx-fileserver}") String gridFsBucketName) {
+		this.repository =  new AASXFileServerMongoDBCrudRepository(template,collectionName,gridFsBucketName);
```

Comment:
Add a new EntityMapping for Package to the BasyxMongoMappingContext component. See AasMongoDBBackendProvider for an example.